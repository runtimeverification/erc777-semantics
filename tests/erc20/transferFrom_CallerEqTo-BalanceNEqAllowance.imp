test.preamble();

// Testing transferFrom(From,To,Value) when To = caller.
// All conditions are met, so it is expected to pass.

test.approve(9, 40);          // set allowance of caller (7) for 9
test.makeCaller(9);           // switch to 9
test.transferFrom(7, 9, 23);  // transfer 23 from 7 to 9
