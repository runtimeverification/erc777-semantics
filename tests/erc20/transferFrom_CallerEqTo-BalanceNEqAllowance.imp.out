7 is approving allowance 40 for 9
Changing the caller to 9
9 is transferring 23 from 7 to 9
Balance of 7 is 47
Balance of 9 is 113
Allowance of 7 for 9 is 17
